### ISSUE TITLE
Parser crashes on array desctructuring in arrow function parameter

### ISSUE THREAD

**Issue by draivin** [webpack/webpack#3238]
**I'm submitting a bug report**

**Webpack version:**
2.1.0-beta.25

**Please tell us about your environment:**
Windows 10

**Current behavior:**
Webpack Parser crashes on array destructuring in arrow function parameter.
```
TypeError: Cannot read property 'type' of null
    at Parser.enterPattern (...\node_modules\webpack\lib\Parser.js:906:27)
    at Parser.<anonymous> (...\node_modules\webpack\lib\Parser.js:926:8)
    at Array.forEach (native)
    at Parser.enterArrayPattern (...\node_modules\webpack\lib\Parser.js:925:19)
    at Parser.enterPattern (...\node_modules\webpack\lib\Parser.js:907:38)
    at ...\node_modules\webpack\lib\Parser.js:892:18
    at Array.forEach (native)
    at Parser.inScope (...\node_modules\webpack\lib\Parser.js:890:9)
    at Parser.walkArrowFunctionExpression (...\node_modules\webpack\lib\Parser.js:685:7)
    at Parser.walkExpression (...\node_modules\webpack\lib\Parser.js:650:40)
    at Parser.<anonymous> (...\node_modules\webpack\lib\Parser.js:644:9)
    at Array.forEach (native)
```

**Expected/desired behavior:**
The code should parse correctly.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem along with a gist/jsbin of your webpack configuration.** 
```js
    let a = [].find(([, i]) => true);
    console.log(a);
```
 
* **Browser:** all
 
* **Language:** ES6

**Comment by TheLarkInn** [comment]
@sokra is this an acorn bug? Or specific to our parser.

**Comment by Kovensky** [comment]
Acorn can parse it fine: https://astexplorer.net/#/PtsxvTSUfZ

The bug is in webpack's parser, which doesn't expect a (perfectly valid) `null` entry.

**Comment by Kovensky** [comment]
I wrote a fix in #3250, but I'm not sure how to write the tests such that webpack doesn't try to run the syntax through Uglify, or doesn't try to run it in node <= 5. Node 4 doesn't support destructuring.
